DeviceAdapters/DemoCamera/DemoCamera.cpp

```diff
@@ -1289,6 +1301,47 @@ int CDemoCamera::OnTestProperty(MM::PropertyBase* pProp, MM::ActionType eAct, lo
 
 }
 
+void CDemoCamera::SlowPropUpdate(std::string leaderValue)
+{
+      // wait in order to simulate a device doing something slowly
+      // in a thread
+      long delay; GetProperty("AsyncPropertyDelayMS", delay);
+      CDeviceUtils::SleepMs(delay);
+      {
+         MMThreadGuard g(asyncFollowerLock_);
+         asyncFollower_ = leaderValue;
+      }
+      OnPropertyChanged("AsyncPropertyFollower", leaderValue.c_str());
+   }
+
+int CDemoCamera::OnAsyncFollower(MM::PropertyBase* pProp, MM::ActionType eAct)
+{
+   if (eAct == MM::BeforeGet){
+      MMThreadGuard g(asyncFollowerLock_);
+      pProp->Set(asyncFollower_.c_str());
+   }
+   // no AfterSet as this is a readonly property
+   return DEVICE_OK;
+}
+
+int CDemoCamera::OnAsyncLeader(MM::PropertyBase* pProp, MM::ActionType eAct)
+{
+   if (eAct == MM::BeforeGet){
+      MMThreadGuard g(asyncLeaderLock_);
+      pProp->Set(asyncLeader_.c_str());
+   }
+   if (eAct == MM::AfterSet)
+   {
+      std::string asyncLeaderTmp;
+      {
+         MMThreadGuard g(asyncLeaderLock_);
+         pProp->Get(asyncLeader_);
+         asyncLeaderTmp = asyncLeader_;
+      }
```

Comment:
Since you no longer access `asyncLeader_` anywhere else, and since property handlers are called by the Core in a synchronized context, there is no need for `asyncLeaderLock_`; you can remove it entirely.

Reply:
done